MaskRCNN_no_batch_convergence/train.py

```diff
@@ -452,35 +423,46 @@ def roi_heads(self, image, features, proposals, targets):
                                               proposal_boxes)
             if self.training:
                 # Convert nobatch tensors to batch tensors
-                proposal_labels = proposals.labels
-                proposal_gt_id_for_each_fg = [proposals.fg_inds_wrt_gt]
-                batch_gt_boxes = tf.expand_dims(gt_boxes, axis=0)
+                # proposal_labels = proposals.labels
+                # proposal_gt_id_for_each_fg = [proposals.fg_inds_wrt_gt]
                 proposal_fg_inds = tf.reshape(tf.where(proposal_labels > 0), [-1])
                 proposal_fg_boxes = tf.gather(proposal_boxes, proposal_fg_inds)
                 proposal_fg_labels = tf.gather(proposal_labels, proposal_fg_inds)
                 # END Convert nobatch tensors to batch tensors
 
-                fastrcnn_head.add_training_info(batch_gt_boxes,
+                fastrcnn_head.add_training_info(input_gt_boxes,
                                                 proposal_labels,
                                                 proposal_fg_inds,
                                                 proposal_fg_boxes,
                                                 proposal_fg_labels,
                                                 proposal_gt_id_for_each_fg)
-                all_losses = fastrcnn_head.losses(BATCH_SIZE_PLACEHOLDER)
 
-        else:
+                all_losses = fastrcnn_head.losses(BATCH_SIZE_PLACEHOLDER)
 
-            fastrcnn_head = FastRCNNHead(proposals, fastrcnn_box_logits, fastrcnn_label_logits,
-                                     gt_boxes, tf.constant(cfg.FRCNN.BBOX_REG_WEIGHTS, dtype=tf.float32))
-            if self.training:
-                all_losses = fastrcnn_head.losses()
-        ##########################################################################################################
+        else: # BATCH_FRCNN_HEAD = False
 
+            proposals = BoxProposals(proposal_boxes)
```

Comment:
`self.rpn` will return proposal_boxes now instead of `BoxProposals`.  Need to move the constructor further down so the no batch code path still works.